imagepipeline/src/main/java/com/facebook/imagepipeline/producers/ResizeAndRotateProducer.java

```diff
@@ -164,6 +167,22 @@ private void doTransform(EncodedImage encodedImage, boolean isLast) {
           ret.setImageFormat(ImageFormat.JPEG);
           try {
             ret.parseMetaData();
+            int fwidth=ret.getWidth();
+			int fheight=ret.getHeight();
+			float wratio=fwidth/fheight;
+			if (resizeOptions.width<fwidth)
+			{
+				fwidth=fwidth-(fwidth-resizeOptions.width)
+				fheight=fheight-((fwidth-resizeOptions.width)/wratio)
+			}
+			if (resizeOptions.height<fheight)
+			{
+				fheight=fheight-(fheight-resizeOptions.height)
+				fwidth=fwidth-((fheight-resizeOptions.height)*wratio)
+			}
+			ret.setWidth(fwidth);
```

Comment:
This looks wrong. `ret.getWidth()` should reflect the **real** dimensions, not the desired one.